modal/sandbox.py

```diff
@@ -63,6 +64,23 @@ def _validate_exec_args(entrypoint_args: Sequence[str]) -> None:
         )
 
 
+@dataclass(frozen=True)
+class SandboxResourceUsage:
+    """Simple data structure storing resource usage stats of a Sandbox."""
+
+    # CPU usage is in core-nanoseconds. Using 1 core for 1 second is 1e9 core ns.
+    # Using 0.5 core for 1 second is 5e8 core ns.
+    cpu_core_nanosecs: int
```

Comment:
Random ignorable suggestion but `nanosecs` is super non-human-friendly, maybe the dataclass could have a method that presents the underlying data in a more human-readable format to save people some hassle when they're using it interactively or printing it or whatever.

Reply:
Yep good idea, I can do a nice `repr`.

Reply:
Hmm, maybe not, I like this behavior: 

> Return a string containing a printable representation of an object. **For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to [eval()](https://docs.python.org/3/library/functions.html#eval);**

Reply:
Personally I've never thought that notion of `repr()` was very useful, but yeah I think you could do it with `__str__` instead of `__repr__`. I'd probably try to format the CPU/RAM/GPU usage with SI prefixes so it's readable across multiple OOMs.